### ISSUE TITLE
self.log("loss", loss, prog_bar=True) leads to an error

### ISSUE THREAD

**Issue by indigoviolet** [PyTorchLightning/pytorch-lightning#5299]
## 🐛 Bug

There is a confusing inconsistency with the special-casing of `loss`: you normally don't need to log it for it to show up in the progress bar, but you need to log it to use it as a `ModelCheckpoint` `monitor=` key. But if you log it with `prog_bar=True`, an error is thrown.

## Please reproduce using the BoringModel

https://colab.research.google.com/drive/15YTx4akWcl10Zo0VMW6jPSYa4wxZeXac?usp=sharing

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Expected behavior

`loss` should either behave like other metrics (ie. require `self.log` to show up in the progress bar), or its special casing should be consistent (ie. it should be available as a monitor key for ModelCheckpoint automatically). Additionally, "double logging" to progress bar should not cause an error - it should be de-duplicated.

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

**Comment by indigoviolet** [comment]
Here's another really crazy repro: https://colab.research.google.com/drive/1GU3VXdSOOyhd4ZZlXNr4HGLtMrOF62hx

It seems that if you are trying to use `ModelCheckpoint(monitor='loss')`, you need to log `self.log("loss", loss, logger=True)`, but _only_ if you are also logging something else (like "val_loss" in the repro). 

If you have `logger=False` but aren't logging "val_loss", things are fine - no error. 

Either I'm missing something huge, or this is a real mess.

**Comment by aiyolo** [comment]
I think you should use `self.log("train_loss", loss, prog_bar=True)`  or any other key whose name is distinct from 'loss'. The reason behind this error is that you try to show two different "loss" keys on the prog_bar, however, the two 'loss' are calculated in different ways. For more details you can refer to this [issue](https://github.com/PyTorchLightning/pytorch-lightning/issues/5247#issuecomment-750411922)

**Comment by indigoviolet** [comment]
@aiyolo thanks for that, I will use it. 

I still think this behavior is very confusing and should be clarified somehow.

**Comment by awaelchli** [comment]
In this PR  #5620, I overwrite the existing value with what the user logs. Optionally, we could log a warning. Have a look and let me know what you think

**Comment by indigoviolet** [comment]
@awaelchli IMHO, I would try to make the "automatic" loss display behave identical to if the user had already called `self.log("loss", loss, prog_bar=True)`. Specifically, I would want `ModelCheckpoint` to see this `loss` metric. I personally  haven't got much value from the "automatic" loss with the different semantics (some kind of smoothing, I guess), but perhaps others do.

Alternatively, you can error on trying to log `loss` or checkpoint on `loss` and tell users they need to choose a different key, but that seems confusing.

**Comment by awaelchli** [comment]
>  I would try to make the "automatic" loss display behave identical to if the user had already called self.log("loss", loss, prog_bar=True)

Isn't that what I'm doing?

`self.log("loss", loss, prog_bar=True)`
will result in showing the exact value loss in the progress bar.

the smoothing is useful when training is very fast, and the progress bar updates are also very fast.

**Comment by indigoviolet** [comment]
As I understand it, you are making it so that if a user does `self.log("loss", loss, prog_bar=True)`, that will not error out, which is great.

The confusion I'm pointing out  is that when a user sees "loss" being logged automatically, they might think they can use `ModelCheckpoint` with `loss` as a metric, but that will [error out in some situations](https://github.com/PyTorchLightning/pytorch-lightning/issues/5299#issuecomment-752300268) -- in this sense, the automatic logging is not equivalent to `self.log("loss", loss, prog_bar=True)`.

**Comment by awaelchli** [comment]
What appears in the progress bar is distinct form what is logged by design. 

> a user sees "loss" being logged automatically

The loss is not logged automatically. Progress bar display is not logging. It displays the value briefly, then it is gone forever.
If a ModelCheckpoint montors a metric, it must appear in the logged metrics.

**Comment by indigoviolet** [comment]
Since `prog_bar=True` is a flag to `self.log()`, it seems reasonable for a new user's mental model that a metric being displayed in the progress bar is being logged as well => it is eligible be checkpointed on. There are three confusing aspects:

1) loss is being displayed but it can't be checkpointed
2) oh, I can't even log it myself because it conflicts 
3) what is being displayed is not actually what I'm returning as "loss" but some other transformation of it

You are fixing (2), but imo all of this would be better if you named the automatically displayed thing "smoothed_loss" or something similar.
